src/rfactor/rfactor.py

```diff
@@ -1,134 +1,281 @@
-        lst_input = [[file, results_folder] for file in
-                     rainfall_inputdata_folder.iterdir()]
-        if ncpu == -1:
-            ncpu = mp.cpu_count()
-        process_map(single_file, lst_input, max_workers=ncpu)
-
-
-def single_file(lst_inputs):
-    """Compute R-factor based on a single file.
-
-    This function uses Octave to compute the matlab `core.m` function, using
-    one input file as function input.
+    # formula requires mm/hr, intensity is on half an hour
+    return df.rolling(interval, on="datetime")["rain_mm"].sum().max() * 2
+
+
+def _compute_erosivity(
+    rain,
+    intensity_method,
+    event_split=TIME_BETWEEN_EVENTS,
+    event_threshold=MIN_CUMUL_EVENT,
+):
+    """Calculate erosivity according to Verstraeten G. for a single year/station combi
 
     Parameters
     ----------
-    lst_inputs: list
-        List of inputs
+    rain : pd.DataFrame
+        DataFrame with rainfall time series. Need to contain the following columns:
+
+        - *datetime* (pd.Timestamp): Time stamp
+        - *rain_mm* (float): Rain in mm
+    intensity_method : Callable
+        Function to derive the maximal rain intensity (over 30min)
+    event_split : str
+        Time interval to split into individual rain events
+    event_threshold : float
+        Minimal cumulative rain of an event to take into account for erosivity
+        derivation
+
+    Returns
+    -------
+    events : pd.DataFrame
+        DataFrame with erosivity output for each event.
+
+        - *datetime* (pd.Timestamp): Time stamp
+        - *event_rain_cum* (float): Cumulative rain for each event
+        - *max_30min_intensity* (float): Maximal 30min intensity for each event
+        - *event_energy* (float): Rain energy per unit depth for each event
+        - *erosivity* (float): Erosivity for each event
+        - *all_events_cum* (float): Cumulative rain over all events together
+        - *erosivity_cum* (float): Cumulative erosivity over all events together
+
     """
-    filename = lst_inputs[0]
-    path_results = lst_inputs[1]
-
-    check_oct()
-    from oct2py import Oct2Py,Oct2PyError
-    year = filename.stem.split("_")[1]
-    inputdata = np.loadtxt(str(filename.resolve()))
-
-    with Oct2Py() as oc:
-        try:
-            oc.addpath(str(Path(__file__).parent.resolve()))
-            cumEI = oc.core(year, inputdata)
-            filename_out = path_results / (
-                        filename.stem + 'new cumdistr salles.txt')
-            np.savetxt(filename_out, cumEI.T, "%.3f %.2f %.1f")
-            oc.exit()
-        except Oct2PyError as e:
-            raise SystemError(e)
-
-def check_oct():
-    """Check octave installation
+    if ("datetime" not in rain.columns) or ("rain_mm" not in rain.columns):
+        raise RFactorInputException(
+            "DataFrame should contain 'datetime' and 'rain_mm' columns."
+        )
+    # ?TODO -> is the enforcement of a single year required for this implementation
```

Comment:
we should keep this approach (at least with the tags with (years, location))

Reply:
ok, done as such